### PR TITLE
fix: updating bn254 certificate aggregation to provide non signers

### PR DESCRIPTION
**Pull request by bdchatham**
### BN254 Aggregation Fix

  - Fixed threshold calculation: Now uses totalSignerCount (all signers across all digests) instead of mostCommonCount (only the winning digest)
  - Allows participation-based thresholds: System can now require X% total participation while the most common result wins

### ECDSA Simplification

  - Removed unnecessary fields: Eliminated NonSignersPubKeys, AllOperatorsPubKeys, and SignersPublicKeys from certificate
  - Simplified certificate generation
  - Key insight: ECDSA contracts only need signers (signatures are independent), unlike BN254 which needs non-signer subtraction for aggregate cryptography

### Testing

  - Created comprehensive mock tests for both BN254 and ECDSA to verify:
    - Correct threshold calculations
    - Proper contract data formatting
    - ECDSA address sorting in GetFinalSignature
    - Edge cases with multiple digest scenarios

The changes make the code cleaner, more efficient, and correctly distinguish between the different cryptographic requirements of BN254 (aggregate signatures) and ECDSA (independent signatures).

### CODE REVIEW

**Review by eigenmikem — CHANGES_REQUESTED** [review]

**Review by eigenmikem — APPROVED** [review]

File: ponos/pkg/signing/aggregation/bn254.go [hunk 1]
```diff
@@ -332,27 +327,47 @@ func (tra *BN254TaskResultAggregator) VerifyResponseSignature(
...
+		return nil, fmt.Errorf("no signatures for winning digest")
+	}
+
+	// Aggregate only the signatures that signed the winning message
+	var aggregatedSig *bn254.Signature
+	aggregatedPubKey := bn254.NewZeroG2Point()
+
+	for _, signer := range winningGroup.signers {
+		if aggregatedSig == nil {
+			aggregatedSig = signer.signature
```
Comment (gnattishness) [inline ponos/pkg/signing/aggregation/bn254.go:346]: @bdchatham Low priority in my opinion:
This aggregation points the initial value to an existing signature and mutates that.
So one of the original `winningGroup.Signers` would now be mutated so that its `signature` is now the aggregated sig.

Relevant test assertion would be: "running `GenerateFinalCertificate()` twice should return equal `AggregatedBN254Certificate` values.

From what I know, the current program only calls `GenerateFinalCertificate()` once so the bug isn't encountered. (Not urgent) 

Fix is to create a new empty sig to add to. Some options:
1. Have `crypto-libs` implement a `Set()` or `Clone()` method so you can copy the sig.
2. Use crypto-libs `bn254.AggregateSignatures()` on a list.
3. Immediate workaround: `aggregatedSig = bn254.NewSignatureFromBytes(signer.signature.Bytes())`

It might be possible to use the zero value `new(bn254.Signature)` but that seems an unstable implementation detail to not rely on.
